src/coreclr/src/jit/gcencode.cpp

```diff
@@ -3891,33 +3891,71 @@ void GCInfo::gcInfoBlockHdrSave(GcInfoEncoder* gcInfoEncoder, unsigned methodSiz
                 assert(false);
         }
 
-        gcInfoEncoderWithLog->SetGenericsInstContextStackSlot(
-            compiler->lvaToCallerSPRelativeOffset(compiler->lvaCachedGenericContextArgOffset(),
-                                                  compiler->isFramePointerUsed()),
-            ctxtParamType);
+        int offset = 0;
+
+        if (compiler->opts.IsOSR())
+        {
+            CORINFO_PATCHPOINT_INFO* ppInfo = compiler->info.compPatchpointInfo;
+            offset                          = ppInfo->GenericContextArgOffset();
+            assert(offset != -1);
+        }
+        else
+        {
+            offset = compiler->lvaToCallerSPRelativeOffset(compiler->lvaCachedGenericContextArgOffset(),
+                                                           compiler->isFramePointerUsed());
+        }
+
+        gcInfoEncoderWithLog->SetGenericsInstContextStackSlot(offset, ctxtParamType);
     }
     // As discussed above, handle the case where the generics context is obtained via
     // the method table of "this".
     else if (compiler->lvaKeepAliveAndReportThis())
     {
         assert(compiler->info.compThisArg != BAD_VAR_NUM);
-        gcInfoEncoderWithLog->SetGenericsInstContextStackSlot(
-            compiler->lvaToCallerSPRelativeOffset(compiler->lvaCachedGenericContextArgOffset(),
-                                                  compiler->isFramePointerUsed()),
-            GENERIC_CONTEXTPARAM_THIS);
+
+        int offset = 0;
+
+        if (compiler->opts.IsOSR())
+        {
+            CORINFO_PATCHPOINT_INFO* ppInfo = compiler->info.compPatchpointInfo;
+            offset                          = ppInfo->GenericContextArgOffset();
+            assert(offset != -1);
+        }
+        else
+        {
+            offset = compiler->lvaToCallerSPRelativeOffset(compiler->lvaCachedGenericContextArgOffset(),
+                                                           compiler->isFramePointerUsed());
+        }
+
+        gcInfoEncoderWithLog->SetGenericsInstContextStackSlot(offset, GENERIC_CONTEXTPARAM_THIS);
     }
 
     if (compiler->getNeedsGSSecurityCookie())
     {
         assert(compiler->lvaGSSecurityCookie != BAD_VAR_NUM);
 
         // The lv offset is FP-relative, and the using code expects caller-sp relative, so translate.
+        int offset = compiler->lvaGetCallerSPRelativeOffset(compiler->lvaGSSecurityCookie);
+
+        if (compiler->opts.IsOSR())
+        {
+            // The offset computed above already includes the OSR frame adjustment, plus the
+            // pop of the "pseudo0RA" from the OSR frame.
```

Comment:
pseudo0RA ?

Reply:
Should be "pseudo RA".

Refers to extra stack slot the runtime sets up right before transitioning so that the OSR method is entered with the expected misaligned SP, so that OSR methods can use most of the normal prolog/epilog code sequences.